Support/DebugVisualizers/MSVC/SCMSVC.natvis

```diff
@@ -1,23 +1,34 @@
 <?xml version="1.0" encoding="utf-8"?>
 <AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+  <Type Name="SC::SegmentItems&lt;*&gt;">
+    <DisplayString>{{ size={sizeBytes / sizeof($T1)}, capacity={capacityBytes / sizeof($T1)} }}</DisplayString>
+    <Expand>
+      <Item Name="[size]">sizeBytes / sizeof($T1)</Item>
+      <Item Name="[capacity]">capacityBytes / sizeof($T1)</Item>
+    </Expand>
+  </Type>
   <Type Name="SC::Vector&lt;*&gt;">
-    <DisplayString>{{ size={reinterpret_cast&lt;const SegmentHeader*&gt;(reinterpret_cast&lt;const uint8_t*&gt;(items) - sizeof(SegmentHeader))->sizeBytes / sizeof(*items)}, capacity={reinterpret_cast&lt;const SegmentHeader*&gt;(reinterpret_cast&lt;const uint8_t*&gt;(items) - sizeof(SegmentHeader))->capacityBytes / sizeof(*items)} }}</DisplayString>
+    <DisplayString>{ *(SegmentItems&lt;$T1&gt;*)((uint8_t*)items - sizeof(SegmentHeader)) }</DisplayString>

```

Comment:
With VS2022 this could be replaced by this:
```xml
<Intrinsic Name="getSegmentItems" Expression="(SegmentItems&lt;$T1&gt;*)((uint8_t*)items - sizeof(SegmentHeader))" />
<DisplayString>{ getSegmentItems() }</DisplayString>
```

and then you can call `getSegmentItems()` wherever you need (including doing things like `getSegmentItems()->size()` if you define size in SegmentItems.

Reply:
I didn't know that, that's a super neat feature! 
I'm totally fine with a VS2022 only natvis.
Worst case we can resurrect the old one from history and give it a different name if someone requests it loudly enough 😄

Reply:
I can merge it as is or wait for you to change it to use the Intrinsic, as you prefer.